src/sbvr-api/sbvr-utils.ts

```diff
@@ -1166,9 +1166,13 @@ const getIdField = (
 		| 'resourceName'
 		| 'vocabulary'
 	>,
-) =>
-	// TODO: Should resolveSynonym also be using the finalAbstractSqlModel?
-	getFinalAbstractSqlModel(request).tables[resolveSynonym(request)].idField;
+) => {
+	const resolvedSynonym = resolveSynonym(request);
+	const translatedTable = getAbstractSqlModel(request).tables[resolvedSynonym];
+	const resourceName =
+		translatedTable != null ? translatedTable.resourceName : resolvedSynonym;
```

Comment:
What happens if we have `v3TranslatedTable.resourceName == 'v4ResourceName'`, why do we not need to progressively go via each translation to find the final table?

Reply:
My understanding is that this works in the case because the only place where we actually call `getIdField` is if `addReturning && request.affectedIds == null`, which means `affectedIds` hasn't been populated yet, which as far as I tested always happens on the last translation to the DB, so this case won't happen, but none (the current & PRed version) of getIdField is really generic